source/particle/output/ascii.cc

```diff
@@ -106,19 +99,15 @@ namespace aspect
             output << "\n";
           }
 
-        file_index++;
+        this->increment_file_index ();
 
-        return output_file_prefix;
+        return "txt";
```

Comment:
I also am not fond of the return string but in essence, the return stings are being collected and formatted to produce a nice readable string, ie.

```
*** Timestep 0:  t=0 seconds
   Skipping temperature solve because RHS is zero.
   Solving C_1 system ... 0 iterations.
   Rebuilding Stokes preconditioner...
   Solving Stokes system... 34+0 iterations.

   Postprocessing:
     Writing particle output: output/particles/particles-00000.{txt,vtu,h5}
```

I have updated the documentation in all of the header files in `include/aspect/particle/output` directory. Primarily, I specify that the return type of the function 
```
         virtual
          std::string
          output_particle_data(const std::multimap<types::LevelInd, Particle<dim> >     &particles,
                               const Property::ParticlePropertyInformation &property_information,
                               const double current_time);
```
is a string that contains the file extension, i.e.:
```
           * @return The name of the file extension that was written, or any other
           * information that describes what output was produced if for example
           * multiple files were created.
           */
```